metagraph-katana-plugin/metagraph_katana/plugins/katana_for_mg/types.py

```diff
@@ -0,0 +1,82 @@
+from metagraph.plugins.core.types import Graph #, GraphSageNodeEmbedding
+from metagraph.plugins.core.wrappers import GraphWrapper #,GraphSageNodeEmbeddingWrapper
+from .. import has_katana
+from typing import Set, List, Dict, Any
+import copy
+import math
+import numpy as np
+
+
+def _determine_array_dtype(array: np.ndarray) -> str:
+    if array.dtype.char in np.typecodes["AllFloat"]:
+        dtype_str = "float"
+    elif array.dtype.char == np.dtype("bool"):
+        dtype_str = "bool"
+    elif array.dtype.char in np.typecodes["AllInteger"]:
+        dtype_str = "int"
+    else:
+        raise TypeError(
+            f"unable to determine dtype, array.dtype.char={array.dtype.char}"
+        )
+    return dtype_str
+
+
+if has_katana:
```

Comment:
Why would this ever be invoked without katana?

Reply:
This follows the style for all plugin examples they provided (below is an example). For other problems, I will resolve them today or tmr. Thanks Arthur for the quick followup!

 https://github.com/metagraph-dev/metagraph-stellargraph/blob/main/metagraph_stellargraph/plugins/stellargraph/types.py#L24

Reply:
Can you figure out *why* they do it this way in the other plugins? I don't want to follow this odd pattern without being able to document the reason for it. Just because other people write code this way isn't a great argument for doing it that way. ;-)